src/store/domains/globalSearch.js

```diff
@@ -0,0 +1,83 @@
+
+import { makeAutoObservable } from 'mobx';
+
+
+export default class GlobalSearch {
+  uiStore;
+
+  globalSearchAPI;
+
+  allFieldsTerm = '';
+
+  loading = false;
+
+  results = {
+    graphics: [],
+    paintings: [],
+    archivals: [],
+  };
+
+  error = null;
+
+  constructor(uiStore, globalSearchAPI) {
+    makeAutoObservable(this);
+
+    this.uiStore = uiStore;
+    this.globalSearchAPI = globalSearchAPI;
+  }
+
+
+  /* Computed */
+
+  get flattenedSearchResultItems() {
+    const { graphics, paintings, archivals } = this.results;
+    return [...graphics, ...paintings, ...archivals];
+  }
+
+
+  /* Actions */
+
+  setAllFieldsTerm(allFieldsTerm) {
+    this.allFieldsTerm = allFieldsTerm;
+  }
+
+  setSearchLoading(loading) {
+    this.loading = loading;
+  }
+
+  setSearchResults(results) {
+    this.results = results;
+  }
+
+  resetSearchResults() {
+    this.results = {
+      graphics: [],
+      paintings: [],
+      archivals: [],
+    };
+  }
+
+  setSearchFailed(error) {
+    this.error = error;
+  }
+
+  searchForAllFieldsTerm(allFieldsTerm) {
+    const { lang } = this.uiStore;
+
+    this.setAllFieldsTerm(allFieldsTerm);
+
+    if (allFieldsTerm.trim() === '') {
+      this.resetSearchResults();
+      return;
+    }
+
+    this.setSearchLoading(true);
+
+    this.globalSearchAPI.searchGloballyFor(allFieldsTerm, lang).then(
```

Comment:
In dieser Action wird die globalSearchAPI genutzt, um unter Angabe des Suchterms und des Sprachcodes eine Suche anzustoßen. Als Resultat werden die Ergebnisse entweder per `setSearchResults`-Action im Store gesetzt, oder im Fehlerfall die Fehlermeldung im Store festgehalten.

Innerhalb von Actions kann man somit auch andere Actions aufrufen.